.github/workflows/main.yml

```
@@ -24,6 +24,20 @@ jobs:
         with:
           command: build
 
+  build_msrv:
+    name: build with MSRV (1.66.1)
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4.1.1
+      # Re-resolve Cargo.lock with minimal versions
+      - uses: dtolnay/rust-toolchain@nightly
```

```suggestion
      - uses: dtolnay/rust-toolchain@stable
```

Using a nightly version might break CI from time to time.

---

Unfortunately we can't since `cargo update -Z minimal-versions` is [still unstable](https://doc.rust-lang.org/cargo/reference/unstable.html#minimal-versions), I think it's an acceptable risk